src/runners/rfqm_worker_runner.ts

```diff
@@ -47,17 +47,24 @@ const RFQM_JOB_COMPLETED_WITH_ERROR = new Counter({
     labelNames: ['address'],
 });
 
-process.on('uncaughtException', (err) => {
-    logger.error(err);
+process.on(
+    'uncaughtException',
+    // see https://github.com/pinojs/pino/blob/master/docs/help.md#exit-logging
+    pino.final(logger, (error, finalLogger) => {
+        finalLogger.error({ error, workerIndex: RFQM_WORKER_INDEX }, 'RFQM worker exiting due to uncaught exception');
+        process.exit(1);
+    }),
+);
+
+process.on('unhandledRejection', (reason, promise) => {
+    const finalLogger = pino.final(logger);
+    finalLogger.error(
+        { reason, promise, workerIndex: RFQM_WORKER_INDEX },
+        'RFQM worker exiting due to unhandled rejection',
+    );
```

Comment:
hmm, why does the syntax differ between this callback and the one above?

Reply:
OOOOooo are you ready to learn some node?

So these two events actually have different signatures. For `unhandledRejection` we were labeling the first parameter `err`but it's NOT actually error, it's `reason`. This case also passes in the promise for which the rejection was not handled. See https://nodejs.org/api/process.html#process_event_unhandledrejection.

For the `uncaughtException` it's more like what you'd expect: https://nodejs.org/api/process.html#process_event_uncaughtexception